packages/walletconnect-signing-manager/src/lib/walletconnect-signing-manager.spec.ts

```diff
@@ -0,0 +1,274 @@
+    });
+
+    it('should throw an error if the signer is not connected', () => {
+      walletConnectMock.signer = undefined;
+      expect(() => signingManager.getExternalSigner()).toThrow(
+        'WalletConnect signer not connected'
+      );
+    });
+  });
+
+  describe('method: onAccountChange', () => {
+    it('should pass the new accounts to the callback, respecting the SS58 format', () => {
+      const accounts = [{ address: address12 }];
+      (walletConnectMock.subscribeAccounts as jest.Mock).mockImplementation(cb => {
+        cb(accounts);
+      });
+
+      const callback = jest.fn();
+      signingManager.onAccountChange(callback, false);
+
+      expect(callback).toHaveBeenCalledWith([address42]);
+    });
+
+    it('should pass the new accountsWithMetadata to the callback, respecting the SS58 format', () => {
+      const accounts = [{ address: address12 }];
+      (walletConnectMock.subscribeAccounts as jest.Mock).mockImplementation(cb => {
+        cb(accounts);
+      });
+
+      const callback = jest.fn();
+      signingManager.onAccountChange(callback, true);
+
+      expect(callback).toHaveBeenCalledWith([
+        {
+          address: address42,
+          meta: {
+            genesisHash: undefined,
+            name: undefined,
+            source: 'walletName',
+          },
+          type: undefined,
+        },
+      ]);
+    });
+
+    it('should use default wallet name if a session is not provided', () => {
+      const accounts = [{ address: address12 }];
+      walletConnectMock.session = undefined;
+      (walletConnectMock.subscribeAccounts as jest.Mock).mockImplementation(cb => {
+        cb(accounts);
+      });
+
+      const callback = jest.fn();
+      signingManager.onAccountChange(callback, true);
+
+      expect(callback).toHaveBeenCalledWith([
+        {
+          address: address42,
+          meta: {
+            genesisHash: undefined,
+            name: undefined,
+            source: 'walletConnect',
+          },
+          type: undefined,
+        },
+      ]);
+    });
+
+    it("should throw an error if the Signing Manager doesn't have a SS58 format", async () => {
+      signingManager = await WalletConnectSigningManager.create({
+        appName: 'testDApp',
+        config,
+      });
+      const accounts = [{ address: address12 }];
+      (walletConnectMock.subscribeAccounts as jest.Mock).mockImplementation(cb => {
+        cb(accounts);
+      });
+
+      expect(() => signingManager.onAccountChange(() => console.log('12345678'))).toThrow(
+        "Cannot call 'onAccountChange callback' before calling 'setSs58Format'. Did you forget to use this Signing Manager to connect with the Polymesh SDK?"
+      );
+    });
+  });
+
+  describe('method: getSs58Format', () => {
+    it('should throw an error if the SS58 format has not been set', () => {
+      const methodName = 'testMethod';
+      signingManager = new (WalletConnectSigningManager as any)(walletConnectMock);
```

Comment:
You should be able to test this by `await WalletConnectSigningManager.create()`, which is preferable than casting to `as any`.

Otherwise add the skip comment at least